File: src/org/newtonpolyhedron/entity/ExecutorRunnable.java
package org.newtonpolyhedron.entity;

import java.io.PrintWriter;

public class ExecutorRunnable implements Runnable {
	
	private final SolverPrinter <?>	solver;
	private final PrintWriter		output;
	
	public ExecutorRunnable(final SolverPrinter <?> solverPrinter, final PrintWriter output) {
		this.solver = solverPrinter;
		this.output = output;
	}
	
	@Override
	public void run() {
		try {
			solver.solveAndPrint();
		} catch(final Exception ex) {
			ex.printStackTrace(output);
		}
	}
}

File: src/org/newtonpolyhedron/entity/ExecutorRunnable.scala
package org.newtonpolyhedron.entity

import java.io.PrintWriter

class ExecutorRunnable(solver: SolverPrinter[_], output: PrintWriter) extends Runnable {

  override def run: Unit =
    try
      solver.solveAndPrint
    catch {
      case th: Throwable => th.printStackTrace(output)
    }
}